index.js

```diff
@@ -300,7 +300,17 @@ function rpcType(payload, paramset) {
             } else if (val > paramset.MAX) {
                 val = paramset.MAX;
             }
-            val = {explicitDouble: val};
+            /* JavaScript doesn't seperate integer and float/double types, so in JavaScript there's only "Number".
+             * However, the XML-RPC library needs to determine whether to wrap the value in <i4>22</i$> or <double>22</double>,
+             * see [list of allowed datatypes](https://ws.apache.org/xmlrpc/types.html).
+             *
+             * node-xmlrpc does this with an `if ( value % 1 == 0)`, see [serializer.js:188](https://github.com/baalexander/node-xmlrpc/blob/d9c88c4185e16637ed5a22c1b91c80e958e8d69e/lib/serializer.js#L188)
+             * 
+             * homematic-xmlrpc introduced an object like `{explicitDouble: val}`.
+             * 
+             * The CCU2 seems to accept <string>22</string> messages and does a string to int/float conversion itself - currently in testing.
```

Comment:
Hmm... I'm asking myself if this works with the binrpc protocol also...

Reply:
That's a good question, because I never checked.
Funny thing is - it does 😄

``` 
root@vserver:~# docker logs hm2mqtt
2017-12-21 13:05:05.161 <info>  hm2mqtt 2.3.0 starting
2017-12-21 13:05:05.164 <info>  mqtt trying to connect mqtt://10.1.1.50
2017-12-21 13:05:05.243 <info>  mqtt connected mqtt://10.1.1.50
2017-12-21 13:05:05.244 <info>  mqtt subscribe hm/set/#
2017-12-21 13:05:05.245 <info>  mqtt subscribe hm/paramset/#
2017-12-21 13:05:05.245 <info>  mqtt subscribe hm/rega/#
2017-12-21 13:05:05.245 <info>  mqtt subscribe hm/rpc/#
2017-12-21 13:05:05.245 <info>  mqtt subscribe hm/command/#
2017-12-21 13:05:05.257 <info>  rpc rfd > init [ 'xmlrpc_bin://10.1.1.50:2127', 'hm2mqtt_rfd' ]
2017-12-21 13:05:05.261 <info>  rpc hmip > init [ 'http://10.1.1.50:2126', 'hm2mqtt_hmip' ]
2017-12-21 13:05:05.750 <error> unknown device hm2mqtt_rfd LEQ1519968:0
2017-12-21 13:05:05.750 <error> unknown device hm2mqtt_rfd NEQ0629891:0
2017-12-21 13:05:05.751 <error> unknown device hm2mqtt_rfd NEQ0879789:0
2017-12-21 13:05:05.751 <error> unknown device hm2mqtt_rfd NEQ0880952:0
2017-12-21 13:05:05.751 <error> unknown device hm2mqtt_rfd NEQ0881133:0
2017-12-21 13:05:10.588 <error> createParamsetQueue called for unknown devices hmip
2017-12-21 13:05:10.962 <info>  rfd got 135 devices and channels
2017-12-21 13:05:34.504 <error> unknown paramsetDescription HM-TC-IT-WM-W-EU/22/THERMALCONTROL_TRANSMIT
2017-12-21 13:05:34.505 <error> unknown paramsetDescription HM-TC-IT-WM-W-EU/22/THERMALCONTROL_TRANSMIT
2017-12-21 13:05:34.505 <error> unknown paramsetDescription HM-TC-IT-WM-W-EU/22/THERMALCONTROL_TRANSMIT
2017-12-21 13:05:40.419 <info>  got 22 paramsetDescriptions
2017-12-21 13:15:05.294 <info>  rpc hmip > init [ 'http://10.1.1.50:2126', 'hm2mqtt_hmip' ]
2017-12-21 13:15:10.603 <error> createParamsetQueue called for unknown devices hmip
2017-12-21 13:25:05.320 <info>  rpc hmip > init [ 'http://10.1.1.50:2126', 'hm2mqtt_hmip' ]
2017-12-21 13:25:10.637 <error> createParamsetQueue called for unknown devices hmip
2017-12-21 13:35:05.328 <info>  rpc hmip > init [ 'http://10.1.1.50:2126', 'hm2mqtt_hmip' ]
```

I started the script a few weeks ago with my most recent branch. It used `xmlrpc_bin` as protocol and yeah.. works since then. For e.g. `hm/set/NEQ0881133_4/SET_TEMPERATURE` accepts either `19` and `19.5` doing the string conversion.